Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionMapsView.cs

```diff
@@ -18,10 +18,15 @@ public ActionMapsView(VisualElement root, StateContainer stateContainer)
             m_ListView = root.Q<ListView>("action-maps-list-view");
             m_ListView.selectionType = UIElements.SelectionType.Single;
 
+            // Setup a selection change filter that prevents the user from deselecting items.
+            // This is desirable since deselecting an Action Map (ESC) would make the Action panel empty with
+            // the current UI layout. Note that we also avoid dispatching select commands if callback
+            // is indirectly triggered by a redraw (instead of constantly hooking and unhooking listener).
             m_ListViewSelectionChangeFilter = new CollectionViewSelectionChangeFilter(m_ListView);
             m_ListViewSelectionChangeFilter.selectedIndicesChanged += (selectedIndices) =>
             {
-                Dispatch(Commands.SelectActionMap((string)m_ListView.selectedItem));
+                if (!isRedrawInProgress)
+                    Dispatch(Commands.SelectActionMap((string)m_ListView.selectedItem));
```

Comment:
The only purpose with this is when Selection Change is coming from the State which triggers a Redraw, which in turn updates m_ListView selection. In that case Dispatch here would only schedule a command that would lead to the very same update, hence yet an unnecessary update. isRedrawInProgress would only be true if this code is triggered from within Redraw (since single threaded). Hence this wouldn't block selection changes to m_ListView triggered by the user. Doing something really fast shouldn't matter (but could get throttled by UI framework) since single threaded and operating from a queue. Or might I be wrong about these expectations? At least this is what I saw when doing these changes on my Mac.

Reply:
As far as I know the redraw is triggered at two places: after we Dispatch() a command and after we change the serializedObject of the state. Interesting is that this happens twice as soon as we do both? So if we change a SerializedObject and do a (eg.) selection after (in the same command) there is a possibility that we abort the second redraw that is actually necessary. Maybe I am missing something, it seems to be working, hopefully it is not just good timing.